drltrace/scripts/headers_parser.py

```diff
@@ -0,0 +1,176 @@
+# ***************************************************************************
+# Copyright (c) 2017 Google, Inc.  All rights reserved.
+# ***************************************************************************
+#
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice,
+#   this list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of Google, Inc. nor the names of its contributors may be
+#   used to endorse or promote products derived from this software without
+#   specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL GOOGLE, INC. OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+# DAMAGE.
+#
+# ********************************************************************************
+# The script is used to search for WinAPI function prototypes in the SDK's headers
+
+import os
+import sys
+from os import listdir
+from os.path import isfile, join
+
+# the list of ignored expored functions
+common_dll_names = {"DllCanUnloadNow", "DllGetClassObject", "DllRegisterServer",
+                    "DllUnregisterServer", "DllEntryPoint", "DllInstall"}
+
+# the path where a list or lists of WinAPI functions to search for are located
+EXPORT_PATH = "exports\\"
+
+# the path where Windows headers are located (install Microsoft SDK to have them)
+HEADERS_PATH = "C:\\Program Files\\Microsoft SDKs\\Windows\\v7.1\\Include\\"
```

Comment:
This needs to be parameterized and generalized: at least have a FIXME or XXX about this single hardcoded path being temporary